### PR TITLE
[IMP] mail: convert isHistory, isNeedAction and isStarred to compute

### PR DESCRIPTION
**Pull request by phenix-factory**

### CODE REVIEW

File: addons/mail/models/mail_message.py [hunk 1]
```diff
@@ -940,10 +949,11 @@ def message_format(self, format_reply=True):
             message_sudo = self.browse(vals['id']).sudo().with_prefetch(self.ids)
             notifs = message_sudo.notification_ids.filtered(lambda n: n.res_partner_id)
             vals.update({
-                'needaction_partner_ids': notifs.filtered(lambda n: not n.is_read).res_partner_id.ids,
-                'history_partner_ids': notifs.filtered(lambda n: n.is_read).res_partner_id.ids,
+                'needaction_partner_ids': [{'id': p.id} for p in notifs.filtered(lambda n: not n.is_read).res_partner_id],
+                'history_partner_ids': [{'id': p.id} for p in notifs.filtered(lambda n: n.is_read).res_partner_id],
                 'is_note': message_sudo.subtype_id.id == note_id,
                 'is_discussion': message_sudo.subtype_id.id == com_id,
+                'starred_partner_ids': [{'id': p.id} for p in message_sudo.starred_partner_ids],
```
Comment (alexkuhn) [inline addons/mail/models/mail_message.py:956]: Would rename fields to `needactionPartners`, `historyPartners`, `starredPartners`.

File: addons/mail/static/src/models/message.js [hunk 2]
```diff
@@ -574,14 +575,24 @@ registerModel({
...
          */
         isHistory: attr({
-            default: false,
+            compute() {
+                if (!this.messaging.currentPartner) {
+                    return clear();
+                }
+                return this.history_partner_ids.includes(this.messaging.currentPartner);
+            },
         }),
```
Comment (alexkuhn) [inline addons/mail/static/src/models/message.js:584]: Missing `default: false` (applies to all `isX` fields in diff)

File: addons/mail/static/src/models/messaging_notification_handler.js [hunk 3]
```diff
@@ -505,19 +505,19 @@ registerModel({
         /**
          * @private
          * @param {Object} param0
-         * @param {integer[]} param0.message_ids
+         * @param {Object[]} param0.message_ids
```
Comment (alexkuhn) [inline addons/mail/static/src/models/messaging_notification_handler.js:508]: `param0.message_ids` => `param0.messages`

**Review by alexkuhn — COMMENTED** [review]
LGTM otherwise

File: addons/mail/models/mail_message.py (hunk 1 above)
Comment (phenix-factory) [reply addons/mail/models/mail_message.py:968]: Will do the rename later, in the new message format. Legacy format should use old stuff/convention.

**Pull request closed without merging**